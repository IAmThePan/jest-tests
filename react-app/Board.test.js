import React from 'react';
import { Board } from './cmpts';
import renderer from 'react-test-renderer';

test('Snapshot Tests for Board', () => {
  const component = renderer.create(<Board onClick={() => {}} squares={["X","O",,"x","o",,"This is a test","I am anoather test","Last test"]} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshot Tests for Board where things dont go right', () => {
  expect(() => {
    const component = renderer.create(<Board onClick={() => {}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  }).toThrow();
});
