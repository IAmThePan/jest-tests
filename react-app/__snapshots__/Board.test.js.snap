// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests for Board 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      x
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      o
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      This is a test
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      I am anoather test
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      Last test
    </button>
  </div>
</div>
`;
